### PR TITLE
Fix checkbox to set calendar as default

### LINKED ISSUE #550: Checking or unchecking one calendar as default should not automatically assign the same status to all other calendars
Screen record

"Birthdays & Anniversaries" and "Personal" are the standard calendars of Evolution in this example.
I use the latest stable and unmodified Hera version including all updates.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/92263822-checking-or-unchecking-one-calendar-as-default-should-not-automatically-assign-the-same-status-to-all-other-calendars?utm_campaign=plugin&utm_content=tracker%2F60388692&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60388692&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mcclurgm**
Fixes #550 

This keeps the state of the check button to set a calendar as default in the right state. When you open the calendar edit dialog, it always reverts to unchecked, and checking it will tell Calendar to make that calendar the default. If it's not checked, then nothing happens and the default calendar stays as is.

I'm not convinced this is a great design, but it at least fixes the issues we had before. It means that (outside of what appears when you create a new event) there's no indication of which is default. But as far as I can tell, this PR just fixes the current design, so any redesign can be done later.

If this gets approved I'll update appdata.xml then to avoid conflicts.

### CODE REVIEW

File: src/SourceDialog/SourceDialog.vala [hunk 1]
```diff
@@ -246,13 +242,15 @@ public class Maya.View.SourceDialog : Gtk.Grid {
...
             create_button.set_label (_("Create Calendar"));
+            is_default_check.active = false;
         } else {
             event_type = EventType.EDIT;
             create_button.set_label (_("Save"));
             name_entry.text = source.display_name;
             type_combobox.sensitive = false;
             type_combobox.set_active (0);
             list_store.foreach (tree_foreach);
+            is_default_check.active = false;
```
Comment (danrabbit) [inline src/SourceDialog/SourceDialog.vala:253]: Is this meant to be false on both sides of this conditional?

Comment (mcclurgm) [reply src/SourceDialog/SourceDialog.vala:253]: No, it’s supposed to be this way. That’s what the last blurb in the main comment was about. It’s always unchecked, so I guess it’s something more like a “do I set this as default when the dialog exits“ check than a “default calendar” check. 

I don’t know of a good way to get the status of whether a calendar is default from here. Also, if I did make it reflect the default, then what would it mean to uncheck the current default calendar?

Comment (danrabbit) [reply src/SourceDialog/SourceDialog.vala:253]: Ah okay. Yeah I guess it should probably reflect the correct state on open and then if it's checked we could make it insensitive so you can't uncheck, only check

Comment (mcclurgm) [reply src/SourceDialog/SourceDialog.vala:253]: Oh, that’s a good idea. I’ll see what I can do!

Comment (mcclurgm) [reply src/SourceDialog/SourceDialog.vala]: Alright, here's my take at implementing that. It's much nicer this way!

Comment (mcclurgm) [reply src/SourceDialog/SourceDialog.vala]: One design question I suppose. Should I change the label to something like "Calendar is already the default" to show why it's insensitive, or do you think this is enough?

File: src/SourceDialog/SourceDialog.vala [hunk 2]
```diff
@@ -250,9 +251,19 @@ public class Maya.View.SourceDialog : Gtk.Grid {
             type_combobox.sensitive = false;
             type_combobox.set_active (0);
             list_store.foreach (tree_foreach);
-            is_default_check.active = false;
-            var cal = (E.SourceCalendar)source.get_extension (E.SOURCE_EXTENSION_CALENDAR);
 
+            try {
+                var registry = new E.SourceRegistry.sync (null);
+                var source_is_default = source.equal (registry.default_calendar);
+                debug (@"source_is_default: $source_is_default");
```
Comment (danrabbit) [inline src/SourceDialog/SourceDialog.vala:258]: Typically we use printf style:
```suggestion
                debug ("source_is_default: %b", source_is_default);
```

**Review by danrabbit — CHANGES_REQUESTED** [review]
Seems to work as expected, just one tiny code style thing :)

File: src/SourceDialog/SourceDialog.vala (hunk 2 above)
Comment (mcclurgm) [reply src/SourceDialog/SourceDialog.vala]: Oops, that wasn't even supposed to be there. Deleted it.

File: src/SourceDialog/SourceDialog.vala (hunk 1 above)
Comment (danrabbit) [reply src/SourceDialog/SourceDialog.vala]: We can probably revisit that later if after some use it turns out it's confusing

**Review by danrabbit — APPROVED** [review]

**Comment by mcclurgm** [comment]
Since you approved, while it's running the build, I'll update appdata real quick for this fix.

**Comment by mcclurgm** [comment]
Done. Thanks for your suggestions!

**Comment by 4jNsY6fCVqZv** [comment]
@mcclurgm Thanks for solving the issue!

**Pull request merged**